app/controllers/album.js

```diff
@@ -1,7 +1,48 @@
-const getUserForToken = require('../services/jwt').getUserForToken,
-  albumsService = require('../services/albums');
+const jwt = require('../services/jwt'),
+  errors = require('../errors'),
+  PurchasedAlbum = require('../models').PurchasedAlbum,
+  albumsService = require('../services/albums'),
+  errorMessages = require('../errors').errorMessages;
 
 exports.listAlbums = (req, res, next) =>
-  getUserForToken(req.headers.token)
+  jwt
+    .getUserForToken(req.headers.token)
     .then(user => albumsService.getAlbums().then(r => res.status(200).send(r)))
+    .catch(e => next(errors.resourceNotFound(errorMessages.albumsNotAvailable)));
+
+exports.purchaseAlbum = (req, res, next) =>
+  jwt
+    .getUserForToken(req.headers.token)
+    .then(user =>
+      albumsService.getAlbum(req.params.id).then(album =>
+        PurchasedAlbum.find({ where: { albumId: album.id, userId: user.id } }).then(existingAlbum => {
+          if (existingAlbum) throw errors.badRequest(errorMessages.albumAlreadyPurchased);
+          else {
+            return PurchasedAlbum.createModel({ albumId: album.id, userId: user.id }).then(
+              indbPurchasedAlbum => {
+                res
+                  .status(201)
+                  .send(indbPurchasedAlbum)
+                  .end();
+              }
+            );
+          }
+        })
+      )
+    )
     .catch(next);
+
+exports.listPurchasedAlbums = (req, res, next) =>
+  jwt.getUserForToken(req.headers.token).then(user => {
```

Comment:
In general, when one list is required, is paged. Add page, limit. For the db, check `findAndCountAll`